pkg/segment/query/queryrefresh.go

```diff
@@ -250,6 +253,76 @@ func populateMicroIndices(smFile string) error {
 	return nil
 }
 
+func syncSegMetaWithSegFullMeta(myId uint64) {
+	vTableNames, err := virtualtable.GetVirtualTableNames(myId)
+	if err != nil {
+		log.Errorf("syncSegMetaWithSegFullMeta: Error in getting vtable names, err:%v", err)
+		return
+	}
+
+	allSmi := make([]*segmetadata.SegmentMicroIndex, 0)
+
+	for vTableName := range vTableNames {
+		streamid := utils.CreateStreamId(vTableName, myId)
+		vTableBaseDir := config.GetBaseVTableDir(streamid, vTableName)
+
+		filesInDir, err := os.ReadDir(vTableBaseDir)
+		if err != nil {
+			log.Errorf("syncSegMetaWithSegFullMeta: Error in reading directory, vTableBaseDir:%v , err:%v", vTableBaseDir, err)
+			continue
+		}
+
+		for _, file := range filesInDir {
+			fileName := file.Name()
+			segkey := filepath.Join(vTableBaseDir, fileName, fileName)
```

Comment:
nit: This seems a little odd. I think you're doing this because if the baseDir ends in `/suffix` then the segkey ends in `/suffix/suffix`. Hopefully at some point we'll fix that. Here, I think it would be good to call some function that computes the segkey based on the baseDir